packages/peregrine/lib/talons/CheckoutPage/useShippingMethod.js

```diff
@@ -69,13 +72,47 @@ export const useShippingMethod = props => {
     );
     const [isBackgroundAutoSelecting, setIsBackgroundAutoSelecting] = useState(false);
     const [isUpdateMode, setIsUpdateMode] = useState(false);
-    const [selectedShippingMethod, setSelectedShippingMethod] = useState(null);
-    const [shippingMethods, setShippingMethods] = useState([]);
 
     const hasData =
         data &&
         data.cart.shipping_addresses.length &&
         data.cart.shipping_addresses[0].selected_shipping_method;
+    
+    const derivedPrimaryShippingAddress = data && data.cart.shipping_addresses && data.cart.shipping_addresses.length
+        ? data.cart.shipping_addresses[0]
+        : null;
+    
+    const derivedSelectedShippingMethod = derivedPrimaryShippingAddress
+        ? addSerializedProperty(derivedPrimaryShippingAddress.selected_shipping_method)
+        : DEFAULT_SELECTED_SHIPPING_METHOD;
+
+    // Determine the component's display state.
+    const nextDisplayState = derivedSelectedShippingMethod
+        ? displayStates.DONE
+        : isLoadingShippingMethods || (isSettingShippingMethod && isBackgroundAutoSelecting)
+        ? displayStates.INITIALIZING
+        : displayStates.EDITING;
+    
+    if (nextDisplayState !== displayState) {
```

Comment:
You don't need this check; if React detects the new state matches the previous [it _should_ skip the update](https://reactjs.org/docs/hooks-reference.html#bailing-out-of-a-state-update).